src/link_prober/LinkProber.cpp

```diff
@@ -313,36 +343,51 @@ void LinkProber::handleRecv(
                 mMuxPortConfig.getPortName() %
                 mMuxPortConfig.getBladeIpv4Address().to_string()
             );
-            if (memcmp(icmpPayload->un.uuid.data, IcmpPayload::getGuidData(),
-                       IcmpPayload::getGuid().size()) == 0) {
+            bool isMatch = (memcmp(icmpPayload->uuid, IcmpPayload::getGuidData(), sizeof(icmpPayload->uuid)) == 0);
+            if (isMatch) {
                 MUXLOGTRACE(boost::format("%s: Matching Guid") % mMuxPortConfig.getPortName());
                 // echo reply for an echo request generated by this/active ToR
                 mRxSelfSeqNo = mTxSeqNo;
                 mLinkProberStateMachine.postLinkProberStateEvent(LinkProberStateMachine::getIcmpSelfEvent());
             } else {
-                // echo reply for an echo request generated by peer ToR
                 mRxPeerSeqNo = mTxSeqNo;
                 mLinkProberStateMachine.postLinkProberStateEvent(LinkProberStateMachine::getIcmpPeerEvent());
-                if (ntohl(icmpPayload->command) == static_cast<uint32_t> (Command::COMMAND_SWITCH_ACTIVE)) {
-                    boost::asio::io_service::strand &strand = mLinkProberStateMachine.getStrand();
-                    boost::asio::io_service &ioService = strand.context();
-                    ioService.post(strand.wrap(boost::bind(
-                        static_cast<void (LinkProberStateMachine::*) (SwitchActiveRequestEvent&)>
-                            (&LinkProberStateMachine::processEvent),
-                        &mLinkProberStateMachine,
-                        LinkProberStateMachine::getSwitchActiveRequestEvent()
-                    )));
+            }
+
+            size_t nextTlvOffset = mTlvStartOffset;
+            size_t nextTlvSize = 0;
+            bool stopProcessTlv = false;
+            while ((nextTlvSize = findNextTlv(nextTlvOffset, bytesTransferred)) > 0 && !stopProcessTlv) {
+                Tlv *nextTlvPtr = reinterpret_cast<Tlv *> (mRxBuffer.data() + nextTlvOffset);
```

Comment:
Here we need to check if tlv offset + length is still with received data range.

Reply:
The check is in `findNextTlv`, if the tlv exceeds the bytes transferred, `nextTlvSize` will set to zero.